src/app/EventManagement.cpp

```
@@ -0,0 +1,867 @@
+    VerifyOrExit(buffer != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    apBufWrapper->mpCurrent = buffer;
+    reader.Init(apBufWrapper);
+    aReader.Init(reader);
+exit:
+    return err;
+}
+
+CHIP_ERROR EventManagement::FetchEventParameters(const TLVReader & aReader, size_t aDepth, void * apContext)
+{
+    CHIP_ERROR err                  = CHIP_NO_ERROR;
+    EventEnvelopeContext * envelope = static_cast<EventEnvelopeContext *>(apContext);
+    TLVReader reader;
+    uint16_t extPriority; // Note: the type here matches the type case in EventManagement::LogEvent, priority section
+    reader.Init(aReader);
+
+    VerifyOrExit(envelope->mNumFieldsToRead > 0, err = CHIP_END_OF_TLV);
+
+    if (reader.GetTag() == TLV::ContextTag(EventDataElement::kCsTag_PriorityLevel))
+    {
+        err = reader.Get(extPriority);
+        SuccessOrExit(err);
+        envelope->mPriority = static_cast<PriorityLevel>(extPriority);
+
+        envelope->mNumFieldsToRead--;
+    }
+
+    if (reader.GetTag() == TLV::ContextTag(EventDataElement::kCsTag_DeltaSystemTimestamp))
+    {
+        err = reader.Get(envelope->mDeltaSystemTime.mValue);
+        SuccessOrExit(err);
+
+        envelope->mNumFieldsToRead--;
+    }
+
+exit:
+    return err;
+}
+
+CHIP_ERROR EventManagement::EvictEvent(CHIPCircularTLVBuffer & apBuffer, void * apAppData, TLVReader & aReader)
+{
+    ReclaimEventCtx * ctx             = static_cast<ReclaimEventCtx *>(apAppData);
+    CircularEventBuffer * eventBuffer = ctx->mpEventBuffer;
+    TLVType containerType;
+    EventEnvelopeContext context;
+    const bool recurse = false;
+    CHIP_ERROR err;
+    PriorityLevel imp = PriorityLevel::Invalid;
+
+    // pull out the delta time, pull out the priority
+    err = aReader.Next();
+    SuccessOrExit(err);
+
+    err = aReader.EnterContainer(containerType);
+    SuccessOrExit(err);
+
+    TLV::Utilities::Iterate(aReader, FetchEventParameters, &context, recurse);
```

add (void) in front and add comment explaining why ignoring error return code is ok here.

---

thanks to point out this, need to add this check after iterate.  
  if (err == CHIP_END_OF_TLV)
    {
        err = CHIP_NO_ERROR;
    }
    SuccessOrExit(err);